src/shared/pipes/filter-by-expression.pipe.spec.ts

```diff
@@ -20,17 +20,17 @@
  * as an Intergovernmental Organization or submit itself to any jurisdiction.
 */
 
-import { FilterByPrefixPipe } from './filter-by-prefix.pipe';
+import { FilterByExpressionPipe } from './filter-by-expression.pipe';
 
-describe('FilterByPrefixPipe', () => {
-  let pipe: FilterByPrefixPipe;
+describe('FilterByExpressionPipe', () => {
+  let pipe: FilterByExpressionPipe;
 
   beforeEach(() => {
-    pipe = new FilterByPrefixPipe();
+    pipe = new FilterByExpressionPipe();
   });
 
-  it('should filter by prefix', () => {
-    let values = ['ab1', 'ab2', 'ac1', 'bc1'];
+  it('should filter by expression', () => {
+    let values = ['ab1', 'ab2', 'xc1', 'zc1'];
```

Comment:
Would be good to test that your regexp filtering works, for example typing c1 should return 'xc1', 'zc1'